OpenRA.Mods.Common/Widgets/LabelWithSelectionWidget.cs

```diff
@@ -0,0 +1,119 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Graphics;
+using OpenRA.Primitives;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets
+{
+	class LabelWithSelectionWidget : LabelWidget
+	{
+		// We don't provide a metrics.yaml property for this as it should really be either (in order of preference):
+		// 1. A color explicitly given to this Widget e.g. from a chrome/**/*/yml file.
+		// 2. A color configured for all LabelWithSelectionWidgets in metrics.yml
+		// 3. A the text color given explcitly to the widget this inherits from.
+		[Translate]
+		public Color? TextColorSelected = ChromeMetrics.GetOrValue<Color?>("TextColorSelected", null);
+		public Color BackgroundColorSelected = ChromeMetrics.Get<Color>("TextBackgroundColorSelected");
+		public Func<Color?> GetTextColorSelected;
```

Comment:
Do we have a use case for making these dynamic?

Reply:
You mean the `Func<Type> PropertyName` stuff or the `Type?` stuff?

I copied the `Func<Type>` stuff from the previous PR under the assumption that it was "they way". I'm not aware of a meaningful usecase for them though.

Should I remove?